src/Socket.c

```diff
@@ -766,7 +772,8 @@ int Socket_continueWrite(int socket)
 		curbuflen += pw->iovecs[i].iov_len;
 	}
 
-	if ((rc = Socket_writev(socket, iovecs1, curbuf+1, &bytes)) != SOCKET_ERROR)
```

Comment:
Here the return value -22 was not detected as failure.
So the random value in `bytes` was used. And this has lead to stack corruption in a following call of writev().

Reply:
I don't think the value in bytes will be random - it is initialised to 0 in Socket_writev, and then not written to unless the return from writev was not -1.